src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/UnusedReferencesTableProvider.DataSource.cs

```diff
@@ -124,7 +124,9 @@ public bool TryGetValue(string keyName, out object? content)
                             break;
                         case UnusedReferencesTableKeyNames.ReferenceName:
                             // It is unnecessary to display the full path to project and assembly files.
-                            content = Path.GetFileNameWithoutExtension(ReferenceUpdate.ReferenceInfo.ItemSpecification);
+                            content = ReferenceUpdate.ReferenceInfo.ReferenceType == ReferenceType.Project
+                                ? Path.GetFileNameWithoutExtension(ReferenceUpdate.ReferenceInfo.ItemSpecification)
```

Comment:
Is it even trimming off anything here? Is it valid to have both `Foo.csproj` and `Foo.vbproj` as project references?

Reply:
Yeah. Let's just ditch it for consistency. Was trying to mimic the Solution Explorer and let the icon do the talking. =)